onnx/defs/controlflow/defs.cc

```diff
@@ -261,10 +261,16 @@ void IfInferenceFunction(InferenceContext& ctx) {
             else_elem_type);
       }
 
-      // merge the 'else' shape information to check it's consistent and
-      // augment the 'if' output if possible
-      mergeInShapeInfo(
-          else_output->tensor_type(), *if_output->mutable_tensor_type());
+      if (checkShapeCompatibility(
+              then_output->tensor_type().shape(),
+              else_output->tensor_type().shape())) {
+        // merge the 'else' shape information to check it's consistent and
+        // augment the 'if' output if possible
+        UnionShapeInfo(
```

Comment:
Just the "UnionShapeInfo(…)" should be sufficient, I think. No need for if-condition or the else-branch.

Reply:
You're right.